internal/gardenclient/client.go

```diff
@@ -257,7 +258,27 @@ func (g *clientImpl) GetConfigMap(ctx context.Context, namespace, name string) (
 	return cm, nil
 }
 
+func (g *clientImpl) GetManagedSeed(ctx context.Context, name string) (*seedmanagementv1alpha1.ManagedSeed, error) {
+	managedSeed := &seedmanagementv1alpha1.ManagedSeed{}
+	key := types.NamespacedName{Name: name, Namespace: "garden"} // Currently, managed seeds are restricted to the garden namespace
+
+	if err := g.c.Get(ctx, key, managedSeed); err != nil {
+		return nil, fmt.Errorf("failed to get managed seed %v: %w", name, err)
+	}
+
+	return managedSeed, nil
+}
+
 func (g *clientImpl) GetSeedClientConfig(ctx context.Context, name string) (clientcmd.ClientConfig, error) {
+	managedSeed, err := g.GetManagedSeed(ctx, name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	if managedSeed != nil {
+		return g.GetShootClientConfig(ctx, managedSeed.Namespace, managedSeed.Name)
```

Comment:
if it returns a notFound error (e.g. gardenlogin-controller-manager is not running) it should continue and fallback to the .login kubeconfig secret

Reply:
I think it is better to always return the same kind of kubeconfig for the same seed. It should be clear to user what he gets for a managed seed. If there is no `gardenlogin-controller-manager` we anyway have a problem.

Reply:
I changed it as @petersutter requested (as he had good arguments) but I kind of also agree with @holgerkoser.
Let's discuss this again.

Reply:
removed fallback as discussed